test/my_assembly_test.pl

```diff
@@ -58,15 +62,45 @@
     ok(scalar @{ $genome_obj->{cdss} } gt 0, "Number of CDSs");
     ok(defined($genome_obj->{mrnas}), "mRNAs array is present");
     ok(scalar @{ $genome_obj->{mrnas} } gt 0, "Number of mRNAs");
+    ok($genome_obj->{scientific_name} eq "Acidilobus sp 7", "Sci name is correct");
+    ok(!defined($genome_obj->{taxon_assignments}), "Taxon assignments is undefined");
 }, "test_annotate_assembly";
 print "Summary for $assembly_obj_name\n";
 
-#
-#	TEST ONE ASSEMBLY -
-#
-print "ASSEMBLYREF = $assembly_ref\n";
+# Test processing a single assembly and saving as a genome while specifing a NCBI taxon ID.
 lives_ok {
+	print("######## Running RAST annotation with Taxon ID ########\n");
+    my $params_copy = { %$params };
+    # this tax ID's species name changed in the 2018-12 NCBI dump and again in the 2019-02 dump
+    # so it is a good test case for making sure the timestamp is passed to the RE correctly.
+    # It depends on the RE containing 2018 NCBI data, which it currently does
+    $params_copy->{ncbi_taxon_id} = 2448083;
+    $params_copy->{relation_engine_timestamp_ms} = 1545000000000;  # epoch ms
+
+	my $ret = &make_impl_call("RAST_SDK.annotate_genome", $params_copy);
+	my $genome_ref = get_ws_name() . "/" . $genome_obj_name;
+	my $genome_obj = $ws_client->get_objects([{ref=>$genome_ref}])->[0]->{data};
+    
+	print "\n\nOUTPUT OBJECT DOMAIN = $genome_obj->{domain}\n";
+	print "OUTPUT OBJECT G_CODE = $genome_obj->{genetic_code}\n";
+
+    ok(defined($genome_obj->{features}), "Features array is present");
+    ok(scalar @{ $genome_obj->{features} } gt 0, "Number of features");
+    ok(defined($genome_obj->{cdss}), "CDSs array is present");
+    ok(scalar @{ $genome_obj->{cdss} } gt 0, "Number of CDSs");
+    ok(defined($genome_obj->{mrnas}), "mRNAs array is present");
+    ok(scalar @{ $genome_obj->{mrnas} } gt 0, "Number of mRNAs");
+    ok($genome_obj->{scientific_name} eq "Metarhizium sp. MJH 2018c", "Sci name is correct");
+    cmp_deeply($genome_obj->{taxon_assignments}, {'ncbi' => '2448083'},
+        "Taxon assignments is correct");
+}, "test_annotate_assembly";
```

Comment:
This could all be done with a single test using Test::Deep's `cmp_deeply`, e.g.

```
my $expected = {
   taxon_assignments => { ncbi => 2448083 },
   scientific_name => 'Metarhizium sp. MJH 2018c',
   features => [ all( $feature_structure ) ],  # specify the format that you expect the features to have
   cdss => [ all( $cdss_structure ) ], # specify format
   ...
};
cmp_deeply $genome_obj, $expected, 'Genome object matches expected structure';
```